pkg/controller/servicebindingrequest/retriever_test.go

```diff
@@ -41,65 +40,66 @@ func TestRetrieverNew(t *testing.T) {
+	t.Run("GetCRKey", func(t *testing.T) {
+		imageName, err := retriever.getCRKey("spec", "imageName")
+		assert.Nil(t, err)
+		assert.Equal(t, "postgres", imageName)
 	})
-	assert.Nil(t, err)
 
-	t.Logf("retriever.data '%#v'", retriever.data)
-	assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_SECRET_USER")
-	assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_SECRET_PASSWORD")
+	t.Run("read", func(t *testing.T) {
+		// reading from secret, from status attribute
+		err := retriever.read("status", "dbCredentials", []string{
+			"urn:alm:descriptor:servicebindingrequest:env:object:secret:user",
+			"urn:alm:descriptor:servicebindingrequest:env:object:secret:password",
+		})
+		assert.Nil(t, err)
 
-	// reading from spec attribute
-	err = retriever.read("spec", "image", []string{
-		"urn:alm:descriptor:servicebindingrequest:env:attribute",
-	})
-	assert.Nil(t, err)
+		t.Logf("retriever.data '%#v'", retriever.data)
+		assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_SECRET_USER")
+		assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_SECRET_PASSWORD")
 
-	t.Logf("retriever.data '%#v'", retriever.data)
-	assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_IMAGE")
+		// reading from spec attribute
+		err = retriever.read("spec", "image", []string{
+			"urn:alm:descriptor:servicebindingrequest:env:attribute",
+		})
+		assert.Nil(t, err)
 
-}
+		t.Logf("retriever.data '%#v'", retriever.data)
+		assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_IMAGE")
 
-func TestRetrieverExtractSecretItemName(t *testing.T) {
-	assert.Equal(t, "user", retriever.extractSecretItemName(
-		"urn:alm:descriptor:servicebindingrequest:env:object:secret:user"))
-}
+	})
 
-func TestRetrieverReadSecret(t *testing.T) {
-	retriever.data = make(map[string][]byte)
+	t.Run("", func(t *testing.T) {
+		assert.Equal(t, "user", retriever.extractSecretItemName(
+			"urn:alm:descriptor:servicebindingrequest:env:object:secret:user"))
+	})
 
-	err := retriever.readSecret("db-credentials", []string{"user", "password"})
-	assert.Nil(t, err)
+	t.Run("ReadSecret", func(t *testing.T) {
+		retriever.data = make(map[string][]byte)
 
-	assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_SECRET_USER")
-	assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_SECRET_PASSWORD")
-}
+		err := retriever.readSecret("db-credentials", []string{"user", "password"})
+		assert.Nil(t, err)
 
-func TestRetrieverStore(t *testing.T) {
-	retriever.store("test", []byte("test"))
-	assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_TEST")
-	assert.Equal(t, []byte("test"), retriever.data["SERVICE_BINDING_DATABASE_TEST"])
-}
+		assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_SECRET_USER")
+		assert.Contains(t, retriever.data, "SERVICE_BINDING_DATABASE_SECRET_PASSWORD")
+	})
 
-func TestRetrieverSaveDataOnSecret(t *testing.T) {
-	err := retriever.saveDataOnSecret()
-	assert.Nil(t, err)
-}
+	t.Run("Store", func(t *testing.T) {
```

Comment:
Should the title string follow the same capitalization than the method? So in this case it would become `store`?

Reply:
I made it lowe case. a829d8538772192f7c58ac44ea3074ee0867e2c5